include/zephyr/drivers/timer/system_timer.h

```diff
@@ -109,19 +109,13 @@ extern void sys_clock_announce(int32_t ticks);
  */
 extern uint32_t sys_clock_elapsed(void);
 
-#if defined(CONFIG_SYS_CLOCK_EXISTS) && \
-	defined(CONFIG_SYSTEM_TIMER_HAS_DISABLE_SUPPORT) || \
-	defined(__DOXYGEN__)
 /**
  * @brief Disable system timer.
  *
  * This function is a no-op if the system timer does not have the capability
  * of being disabled.
  */
 extern void sys_clock_disable(void);
-#else
-static inline void sys_clock_disable(void) {}
-#endif /* CONFIG_SYSTEM_TIMER_HAS_DISABLE_SUPPORT */
```

Comment:
this doesn't violate any rule, it just provides a stub when not available. This is done in many other places, e.g. PM.

Reply:
@gmarull technically it contradicts this rule:
https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-1-conditional-compilation

Reply:
The declaration is provided in any case, just that when not available it results in a stub definition.

Reply:
Fair enough. But the change does actually clean up the code, why add the stub if you can conditionally call the function instead?

Reply:
The same applies to many other subsys, but it is often preferred stubs to e.g. avoid cluttering drivers with ifdeffery. This is a common practice in other projects like Linux Kernel as well. In this particular case, I have no strong opinion, clock API is not great anyway, so change doesn't hurt. But please do not justify the change as a fix for a non-existing violation.